codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/eventstream/EventStreamBaseRequirements.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy.protocols.eventstream
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.rust.codegen.core.smithy.CodegenTarget
+import software.amazon.smithy.rust.codegen.core.smithy.RustSymbolProvider
+import software.amazon.smithy.rust.codegen.core.testutil.EventStreamTestRequirements
+import software.amazon.smithy.rust.codegen.server.smithy.RustCodegenServerPlugin
+import software.amazon.smithy.rust.codegen.server.smithy.ServerCodegenContext
+import software.amazon.smithy.rust.codegen.server.smithy.ServerSymbolProviders
+import software.amazon.smithy.rust.codegen.server.smithy.testutil.ServerTestSymbolVisitorConfig
+import software.amazon.smithy.rust.codegen.server.smithy.testutil.serverTestRustSettings
+import software.amazon.smithy.rust.codegen.server.smithy.testutil.serverTestSymbolProvider
+
+abstract class EventStreamBaseRequirements : EventStreamTestRequirements<ServerCodegenContext> {
+    override fun createCodegenContext(
+        model: Model,
+        symbolProvider: RustSymbolProvider,
+        serviceShape: ServiceShape,
+        protocolShapeId: ShapeId,
+        codegenTarget: CodegenTarget,
+    ): ServerCodegenContext {
+        val settings = serverTestRustSettings()
+        val serverSymbolProviders = ServerSymbolProviders.from(
```

Comment:
Use the `software.amazon.smithy.rust.codegen.server.smithy.testutil.serverTestSymbolProviders` function.

Reply:
Or directly `serverTestCodegenContext`, since this test is only testing with `publicConstrainedTypes` enabled.

Reply:
Thanks for the pointers! I'm not so familiar with the server testing helpers.

Reply:
Fixed in be77fa0df3888633117973eed62a7784ed4df258